### DISCUSSION TITLE
eqMac v1.3.0

### DISCUSSION THREAD

**Discussion by nodeful** [bitgapp/eqMac#572] [General]
# v1.3.0 (Pre-Release)
### New Features & Improvements:
* Volume Mixer - Adjust volume per application (Pro)
* UI Resizing - No more fixed UI width/height (Expert EQ only for now)
* 60FPS Spectrum Analyzer
* Driver now mimics selected Output device name
* macOS Monterey Support
* AutoEQ updated to latest Measurements

### Bug Fixes & Refactors
* Fixed Output switching to Built-in device when switching between Equalizers
* Fixed crash related to Output device not having Mute controls (EVO4 etc)
* Fixed Spatial Audio state not being saved
* Fixed State/Settings being deleted between App updates
* Fixed Tooltips not showing
* Fixed Tooltips positioning when UI Scaling is applied
* Fixed memory leaks
* Fixed many misc crashes

<hr /><em>This discussion was created from the release <a href='https://github.com/bitgapp/eqMac/releases/tag/v1.3.0'>eqMac v1.3.0</a>.</em>

**Comment by blcarson** [comment]
Getting a crash as soon as the app is opened - attaching log below. If there's a better way to attach that or place to put this comment please let me know.
[v130.crash.txt](https://github.com/bitgapp/eqMac/files/7271416/v130.crash.txt)

**Reply by nodeful** [reply]
Thanks, I will look into this ASAP

**Reply by nodeful** [reply]
hey @blcarson it has probably something to do with the new Driver crashing. Could you please open Console.app on your Mac and go to Crash Reports, there you will either see a recent `coreaudiod` or `com.apple.audio.Core-Audio-Driver-Service` crash. Could you please send that report to me? Better if you upload it as a txt file to avoid a massive wall of text :) Thank you!

**Reply by blcarson** [reply]
Yep! See attached. I'm also running macOS 12 beta 8 if that helps...

[crash-report.txt](https://github.com/bitgapp/eqMac/files/7271441/crash-report.txt)

**Reply by nodeful** [reply]
Yeah I'm running that as well at the moment :)

**Reply by blcarson** [reply]
Oh well good - if it's working for you than maybe I have some hope! Let me know if you need anything else to diagnose.

**Reply by nodeful** [reply]
@blcarson did it happen constantly, never went away on consecutive launches? Also was any audio playing at the moment of launch? If so what app was playing audio. Also what Audio Device was selected as your output and at what Sampling Rate? Would be helpful if you sent a Screenshot of the device selected in AudioMIDISetup.app.
Thanks!

**Reply by nodeful** [reply]
Hmm the Crash report isn't detailed enough for me to understand what is happening.
I attached a Debug version of the driver below. If you could, please unzip it and place the `eqMac.driver` directory in your `/Library/Audio/Plug-Ins/HAL` folder and restart CoreAudio with `sudo launchctl kickstart -k system/com.apple.audio.coreaudiod`
[eqMac.driver.zip](https://github.com/bitgapp/eqMac/files/7271531/eqMac.driver.zip)

Run eqMac and send me a crash report for the Core Audio Driver Service if it happens

**Reply by blcarson** [reply]
OK, followed your instructions after making sure everything was uninstalled. Here's what I got:
1. Happened on every launch
2. No audio was playing during launch
3. Screenshot from AudioMIDISetup prior to install:
<img width="655" alt="screenshot 1" src="https://user-images.githubusercontent.com/1023153/135718795-310e3af6-411a-41ea-adfe-f32cc935cd9f.png">

Most of install was typical until I got to this step:

<img width="670" alt="screenshot 2" src="https://user-images.githubusercontent.com/1023153/135718311-f74a7f42-e185-4794-aa31-c9fc8a641508.png">

I chose "Restart Mac" and it immediately restarted my computer. I'm not sure if interrupting the pkg install app at that point may have messed things up but that was something of note. Mac the restarted normally, I saw the app for a brief second both on the desktop and in the menu bar. Here are the new reports:

- [core-audio.txt](https://github.com/bitgapp/eqMac/files/7271539/core-audio.txt)
- [eqMac.txt](https://github.com/bitgapp/eqMac/files/7271540/eqMac.txt)

Let me know if I missed anything.

**Reply by nodeful** [reply]
wait, why did you do a pkg reinstall? You just needed to replace the driver with the one I sent :)

**Reply by blcarson** [reply]
Whoops - looks like I missed the follow up comment, my bad :)

I tried replacing that driver but I get these in response: 

<img width="260" alt="screenshot 6" src="https://user-images.githubusercontent.com/1023153/135719785-3e0b72af-52ba-4cd4-855a-cf2c03446c71.png">

[driver-report.txt](https://github.com/bitgapp/eqMac/files/7271591/driver-report.txt)

Trying to unblock it from Security & Privacy actually opens a finder window to the .driver folder we just replaced. Maybe some sort of permissions issue?

**Reply by nodeful** [reply]
Ah sorry I built it for the x64 Intel Architecture, I've attached a Universal binary below
[eqMac.driver.zip](https://github.com/bitgapp/eqMac/files/7271630/eqMac.driver.zip)

**Reply by blcarson** [reply]
OK when I run `sudo launchctl kickstart -k system/com.apple.audio.coreaudiod`, this pops up:

<img width="260" alt="screenshot" src="https://user-images.githubusercontent.com/1023153/135728345-d580f1f0-3413-41f0-92b2-c73b736a3477.png">

Clicking OK brings this up:

<img width="260" alt="screenshot 1" src="https://user-images.githubusercontent.com/1023153/135728382-a5cf48b4-cb85-4fa1-a229-9432fee3016c.png">

What should I do from here?

**Reply by nodeful** [reply]
Sorry I'm not sure what's going on with your setup. This will be hard to debug as I don't yet have an M1 machine (waiting for those juicy M2 processors at the end of the year).
Can I ask this: Delete the eqMac.driver in the HAL directory and reinstall v1.3.0 and try to start it while playing some music, does it still crash? I think I managed to replicate the issue in a macOS 10.12 Virtual Machine, but only crashes when there's no audio playing 🤔

**Reply by blcarson** [reply]
Yep no worries - it's not a normal setup by far. I'll roll back to last version until 12.0 is out of beta and try again.

**Reply by nodeful** [reply]
Hey @blcarson I've just updated the Assets in this release with the fixes that I hope will resolve the issue with the driver. Could you please try to redownload this pre-release and try it out? Thanks!

**Reply by blcarson** [reply]
I think that did the trick! Anything else you'd like me to test?

**Reply by nodeful** [reply]
That's great news. If you find any other issues please let me know. But I think the Version you've just downloaded will be the release binary, so you won't have to redownload :)
Thanks for your help in chasing the bug down.

**Reply by blcarson** [reply]
You're very welcome

**Comment by hexionaut** [comment]
Hello @nodeful,
Why do I need Rosetta to launch EqMac ? Thought it was M1 compatible when I saw the release note of `eqMac Beta v0.3.7` which says "Apple M1 Chip native support".
Am I doing something wrong ?

**Reply by nodeful** [reply]
You dont need Rosetta to launch eqMac, but you do need it to launch the Installer of eqMac. The packaging software I use still hasn't ported to M1, but the app and the driver are ARM compatible
